### PR TITLE
Added Attribute 'protocol' to Request

### PR DESCRIPTION
**Pull request by Jelenkee**
Hey,
one thing I missed in fastify was a way to easily determine whether the used protocol is `http` or `https`.
So I added that feature analogous to `ip` and `hostname` with the `X-Forwarded-Proto`-header.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -78,6 +78,17 @@ function buildRequestWithTrustProxy (R, trustProxy) {
         }
         return this.headers.host || this.headers[':authority']
       }
+    },
+    protocol: {
+      get () {
+        if (this.headers['x-forwarded-proto']) {
```
Comment (jsumners) [inline lib/request.js:84]: I think this should be a check on `trustProxy`? See https://github.com/fastify/fastify/issues/2577

Comment (Jelenkee) [reply lib/request.js:84]: Not necessary. The calling method is called `buildRequestWithTrustProxy`. So the check for `trustProxy` was made before.

File: types/request.d.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ export interface FastifyRequest<
   ip: string;
   ips?: string[];
   hostname: string;
+  protocol: string;
```
Comment (zekth) [inline types/request.d.ts:31]:
```suggestion
  protocol: 'http' | 'https';
```

**Review by zekth — CHANGES_REQUESTED** [review]

**Review by zekth — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Review by delvedor — APPROVED** [review]
LGTM

**Pull request merged**